pkg/services/cleanup/cleanup.go

```diff
@@ -418,3 +425,133 @@ func (srv *CleanUpService) cleanUpTrashAlertRules(ctx context.Context) {
 		logger.Debug("Cleaned up deleted alert rules", "rows affected", affected)
 	}
 }
+
+// cleanupStaleLBACRules exists to clean up lbac rules that are stale from teams getting deleted as we do not have
+// cascading deletions on teams to delete existing lbac rules
+func (srv *CleanUpService) cleanupStaleLBACRules(ctx context.Context) {
+	logger := srv.log.FromContext(ctx)
+
+	// Get all datasources
+	allDataSources, err := srv.dataSourceService.GetAllDataSources(ctx, &datasources.GetAllDataSourcesQuery{})
+	if err != nil {
+		logger.Error("Failed to get datasources for LBAC cleanup", "error", err)
+		return
+	}
+
+	var totalCleaned int
+	var totalDataSources int
+
+	for _, ds := range allDataSources {
+		if ds.JsonData == nil {
+			continue
+		}
+
+		// Check if datasource has team LBAC rules
+		teamHTTPHeaders, err := datasources.GetTeamHTTPHeaders(ds.JsonData)
+		if err != nil || teamHTTPHeaders == nil {
+			continue
+		}
+
+		totalDataSources++
+
+		// Extract team UIDs and check if teams still exist
+		cleanedRules, removedCount := srv.getLBACRulesForTeamsStillExisting(ctx, teamHTTPHeaders, ds.OrgID)
+
+		if removedCount > 0 {
+			// Update the datasource with cleaned rules
+			err := srv.updateDataSourceLBACRules(ctx, ds, cleanedRules)
+			if err != nil {
+				logger.Error("Failed to update datasource LBAC rules",
+					"datasource", ds.UID, "error", err)
+			} else {
+				totalCleaned += removedCount
+				logger.Debug("Cleaned stale LBAC rules",
+					"datasource", ds.UID, "removed", removedCount)
+			}
+		}
+	}
+
+	if totalCleaned > 0 {
+		logger.Info("Cleaned up stale team LBAC rules",
+			"datasources_processed", totalDataSources,
+			"total_rules_removed", totalCleaned)
+	}
+}
+
+func (srv *CleanUpService) getLBACRulesForTeamsStillExisting(ctx context.Context, teamHeaders *datasources.TeamHTTPHeaders, orgID int64) (*datasources.TeamHTTPHeaders, int) {
+	logger := srv.log.FromContext(ctx)
+	cleanedHeaders := &datasources.TeamHTTPHeaders{Headers: make(map[string][]datasources.TeamHTTPHeader)}
+	removedCount := 0
+
```

Comment:
performance nit: we could add, walk through all teams and make one call to teamService for fetching teams.